### PR TITLE
[Issue-25281]: Adding links for the device selector.

### LINKED ISSUE #25281: Build New Custom Tab Navigation Button & Menu
### Parent Issue

_No response_

### Task

Need to build a new custom tab navigation drop-down component, and menu.

See designs under "External Links."

### Proposed Objective

Please Select

### Proposed Priority

Please Select

### Acceptance Criteria

- [x] Behavior notes… if there are “User Defined” Devices, we’ll show 3 columns. If there are no "User Defined" Devices, we'll just show 2.

 - [ ] If the user has the ability to add content, we’ll also show them the “Add Device” link that will take them to the Content Search view for "Device".

image

- [ ] Remove Legacy “Device” Preview Menu. We’re merging that functionality into the Preview Tab.

image

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

Demo - https://www.figma.com/proto/jyu7b1wgL16iV7yOoNdgu9/SEO?page-id=1%3A65&type=design&node-id=372-9191&viewport=-1440%2C-118%2C0.37&scaling=scale-down&starting-point-node-id=372%3A9190

Menu - https://www.figma.com/file/jyu7b1wgL16iV7yOoNdgu9/SEO?type=design&node-id=428-13077&t=9Qcws8is2LP5BkgC-0

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by manuelrojas**
Refs #25281

### <samp>🤖 Generated by Copilot at 2f5f6b5</samp>

This pull request improves the user experience of the SEO analysis portlet by adding router links to the device selector component. The router links allow the user to open the page preview and the devices portlet from the SEO portlet. It also removes some redundant code and properties related to the page preview from other components. It updates the tests and templates accordingly.

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots

https://github.com/dotCMS/core/assets/3438705/b41eeb4b-688c-49fa-b773-b14a21250f7b

### CODE REVIEW

**Review by zJaaal — COMMENTED** [review]
See my suggestions

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html [hunk 1]
```diff
@@ -9,7 +9,12 @@
                 <h3 class="dot-device-selector__header-title">
                     {{ 'editpage.device.selector.title' | dm }}
                 </h3>
-                <p-button icon="pi pi-plus" styleClass="p-button-rounded p-button-text"></p-button>
+                <a [routerLink]="linkToAddDevice" data-testId="dot-device-link-add">
+                    <p-button
+                        icon="pi pi-plus"
+                        styleClass="p-button-rounded p-button-text"
+                    ></p-button>
+                </a>
```
[the remark this answers is not in the document]
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html:17]: Im not sure if it has the same design.

[the remark this answers is not in the document]
Comment (manuelrojas) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html:17]: @zJaaal checking

[the remark this answers is not in the document]
Comment (manuelrojas) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html:17]: @zJaaal It looks quite different :(

[the remark this answers is not in the document]
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.html:17]: dont worry about it then.

**Review by zJaaal — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.ts [hunk 2]
```diff
@@ -145,4 +149,15 @@ export class DotDeviceSelectorSeoComponent implements OnInit {
...
     }
+
+    @Input()
+    set apiLink(value: string) {
+        if (value) {
+            const frontEndUrl = `${value.replace('api/v1/page/render', '')}`;
+
+            this.previewUrl = `${frontEndUrl}${
+                frontEndUrl.indexOf('?') != -1 ? '&' : '?'
+            }disabledNavigateMode=true`;
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.ts:160]: I think in this case the URL api will work. All this parsing is hard to read.

**Review by fmontes — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**